### PR TITLE
Simplified addVariable and addFactor signatures

### PR DESCRIPTION
**Pull request by Affie**

### CODE REVIEW

**Review by dehann — COMMENTED** [review]
Looks good to me, thanks!  Let's see how it falls out with similar draft in SDK.js.

**Comment by Affie** [comment]
I tested on tutorial 4 and this will be the change:
```python
# current signiture (will stilll work btw.)
f = Factor("x0c0c1c2c3f1", 
           "Pose2Pose2", 
           ["x0", "c0", "c1", "c2", "c3"], 
           FactorData(fnc=Pose2Pose2(Z=FullNormal(mu=[-2.,-2, 0], cov=np.power(np.diag([0.5, 0.5, 0.05]),2))).dump(), 
                      multihypo=[1.0, 0.25, 0.25, 0.25, 0.25])
)
result_id = await addFactor(client, context, f) 

# changed to
result_id = await addFactor(client, context, 
                            ["x0", "c0", "c1", "c2", "c3"], 
                            Pose2Pose2(Z=FullNormal(mu=[-2.,-2, 0], cov=np.power(np.diag([0.5, 0.5, 0.05]),2))), 
                            multihypo=[1.0, 0.25, 0.25, 0.25, 0.25])
```

For me, it's an obvious improvement and close enough to the Ceasar.jl api to meet the similar signatures requirement :
```julia
addFactor!(
    fg, 
    [:x0, :c0, :c1, :c2, :c3], 
    Pose2Pose2(MvNormal([-2, -2, 0.], diagm([0.5, 0.5, 0.05].^2))), 
    multihypo=[1.0, 0.25, 0.25, 0.25, 0.25]
)
```

**Review by GearsAD — APPROVED** [review]

File: src/navability/services/factor.py [hunk 1]
```diff
@@ -33,8 +39,23 @@
...
 
+def assembleFactorName(xisyms: List[str]):
+    s = "".join(xisyms) + 'f_' + str(uuid4())[0:4]
+    return s
+
+def getFncTypeName(fnc: InferenceType):
+    return type(fnc).__name__
+
+def addFactor(client: NavAbilityClient, context: Client, f_labels, fnc=None, multihypo = [], nullhypo=0.0):
+    if isinstance(f_labels, Factor):
```
[the remark this answers is not in the document]
Comment (Affie) [reply src/navability/services/factor.py:50]: Haha, I'm sorry. I'm too used to multiple dispatch. I'll leave the fix up to someone that knows what they are doing :-)

**Comment by Affie** [comment]
I have one more thing that I want us to consider:

To allow 
```python
addVariable(navability_client, client, \"x1\", VariableType.Pose2.value) # ie. "RoME.Pose2"
#and 
addVariable(navability_client, client, \"x1\", VariableType.Pose2)
```

**Comment by dehann** [comment]
I'd say yes, just add a comment in the code that in future we will remove the non-standard calls, e.g. we'd only use `VariableType.Pose2` in future versions.

**Comment by Affie** [comment]
> I'd say yes, just add a comment in the code that in future we will remove the non-standard calls, e.g. we'd only use `VariableType.Pose2` in future versions.

That's my thoughts exactly. I much prefer the concrete (`VariableType.Pose2`) version.
Give me a few minutes and I'll update it.

**Review by dehann — APPROVED** [review]

**Pull request merged**